diff --git a/src/cencode.c b/src/cencode.c
--- a/src/cencode.c
+++ b/src/cencode.c
@@ -7,7 +7,9 @@ For details, see http://sourceforge.net/projects/libb64
 
 #include <b64/cencode.h>
 
+/*
 const int CHARS_PER_LINE = 72;
+*/
 
 void base64_init_encodestate(base64_encodestate* state_in)
 {
@@ -72,12 +74,14 @@ int base64_encode_block(const char* plaintext_in, int length_in, char* code_out,
 			result  = (fragment & 0x03f) >> 0;
 			*codechar++ = base64_encode_value(result);
 			
+			/*
 			++(state_in->stepcount);
 			if (state_in->stepcount == CHARS_PER_LINE/4)
 			{
 				*codechar++ = '\n';
 				state_in->stepcount = 0;
 			}
+			*/
 		}
 	}
 	/* control should not reach here */
@@ -102,7 +106,9 @@ int base64_encode_blockend(char* code_out, base64_encodestate* state_in)
 	case step_A:
 		break;
 	}
+	/*
 	*codechar++ = '\n';
+	*/
 	
 	return codechar - code_out;
 }
