diff --git a/WIN32-Code/event2/event-config.h b/WIN32-Code/event2/event-config.h
--- a/WIN32-Code/event2/event-config.h
+++ b/WIN32-Code/event2/event-config.h
@@ -167,7 +167,9 @@
 /* #undef _EVENT_HAVE_STRSEP */
 
 /* Define to 1 if you have the `strtok_r' function. */
-/* #undef _EVENT_HAVE_STRTOK_R */
+#if defined(__MINGW32__) || defined(__MINGW64__)
+#define _EVENT_HAVE_STRTOK_R 1
+#endif
 
 /* Define to 1 if you have the `strtoll' function. */
 /* #define _EVENT_HAVE_STRTOLL 1 */
